### PR TITLE
Various minor improvements (@icon, suffix-schema configuration)

### PR DESCRIPTION
**Pull request by wu-kelvin**
- Add optional config param (suffix-schema) on both `plugin-terraform`, `plugin-json`. Reason behind this is to support in the future if anyone would like additional suffix to their generated schema
- Fixed sortable:trait to get value instead of object literal
- Added @icon support to be included to the generated terraform
- Updated README and comment
- Added guide in plugin-json `codegen.yml` on how to generate typescript schema by using the `codegen-cli`
- Updated to `graphql 16`
- Added support for dynamic visualization
- Change blocks type of terraform `amplience_slots_repository` to `amplience_content_repository`

### CODE REVIEW

File: packages/plugin-terraform/src/index.ts [hunk 1]
```diff
@@ -100,29 +102,38 @@ export const plugin: PluginFunction<PluginConfig> = (
...
+          validation_level: isSlot ? 'SLOT' : 'CONTENT_TYPE',
         })
 
         const shouldVisualize = maybeDirectiveValue<BooleanValueNode>(
           directive,
           'visualizations'
         )?.value
 
+        const iconUrl = maybeDirectiveValue<EnumValueNode>(
+          maybeDirective(node, 'icon')!,
```
Comment (korsvanloon) [inline packages/plugin-terraform/src/index.ts:119]: Where do you guarantee that icon exists?
I think you should surround this with a hasDirective

```ts
const iconUrl = hasDirective(node, 'icon') ? maybeDire....?.value : undefined
```

File: packages/plugin-terraform/src/index.ts [hunk 2]
```diff
@@ -100,29 +102,38 @@ export const plugin: PluginFunction<PluginConfig> = (
...
             shouldVisualize && visualization
               ? {
                   label: 'Visualization',
                   templated_uri: maybeArg(visualization),
-                  default: true
+                  default: true,
                 }
-              : undefined
+              : undefined,
+          card: [{}], // TODO
```
Comment (korsvanloon) [inline packages/plugin-terraform/src/index.ts:136]: Please either remove this property or implement it :)

**Review by korsvanloon — CHANGES_REQUESTED** [review]

File: packages/plugin-json/examples/codegen.yml [hunk 3]
```diff
@@ -5,8 +5,43 @@ hooks:
...
+      #   config:
+      #     maybeValue: T | undefined
+      #     enumsAsTypes: false
+      #     noNamespaces: false
+      #     preResolveTypes: false
+      #     onlyOperationTypes: false
+      #     skipTypename: false
+      #     scalars:
+      #       Set: string[]
+      #       Time: number
```
[the remark this answers is not in the document]
Comment (wu-kelvin) [reply packages/plugin-json/examples/codegen.yml]: noted !

File: packages/plugin-json/examples/codegen.yml [hunk 4]
```diff
@@ -5,8 +5,43 @@ hooks:
...
+      #       Date: string
+      #       DateTime: string
+      #       Json: "{ [key: string]: any } | string | string[] | { [key: string]: any }[]"
```
[the remark this answers is not in the document]
Comment (wu-kelvin) [reply packages/plugin-json/examples/codegen.yml]: sure, will do it right away

File: packages/plugin-json/package.json [hunk 5]
```diff
@@ -1,31 +1,31 @@
...
-  "version": "0.1.8",
-  "private": false,
-  "license": "MIT",
-  "module": "dist/index.mjs",
-  "main": "dist/index.js",
-  "typings": "dist/index.d.ts",
-  "files": [
-    "dist"
-  ],
+  "dependencies": {
```
[the remark this answers is not in the document]
Comment (wu-kelvin) [reply packages/plugin-json/package.json]: yep, undone all the sorting in the files

File: packages/plugin-json/src/lib/amplience-schema-transformers.ts [hunk 6]
```diff
@@ -34,37 +34,43 @@ export const contentTypeSchemaBody = (
...
-    )
-    .filter(field => field.type.kind === 'NonNullType')
-    .map(n => n.name.value)
-})
-
+      .filter((field) => field.type.kind === 'NonNullType')
+      .map((n) => n.name.value),
+  }
+  const ref = refType(AMPLIENCE_TYPE.CORE.HierarchyNode).allOf[0]
+  if (hierarchy) amplienceContentTypeSchema.allOf.push(ref)
```
[the remark this answers is not in the document]
Comment (wu-kelvin) [reply packages/plugin-json/src/lib/amplience-schema-transformers.ts]: sure, will change to conditional property and inline it. 
For the coding styles, could you share the settings / prettier file that you have so I can reapply the styles, apologize for the changes

File: packages/plugin-terraform/src/index.ts (hunk 1 above)
Comment (wu-kelvin) [reply packages/plugin-terraform/src/index.ts]: yea sure, be better this way, done !

File: packages/plugin-terraform/src/index.ts (hunk 2 above)
Comment (wu-kelvin) [reply packages/plugin-terraform/src/index.ts]: will remove it, and implement it later once we get into the card topic

**Pull request merged**